python312Packages.{jax,jaxlib}: 0.4.28 -> 0.4.38; use jaxlib-bin by default

natsukium opened the pull request:
- python312Packages.jaxlib: use jaxlib-bin by default
Thanks to bazel, it is difficult to source build with our current resources.

- python312Packages.jaxlib-bin: 0.4.28 -> 0.4.36
Diff: https://github.com/jax-ml/jax/compare/jaxlib-v0.4.28...jaxlib-v0.4.38
Changelog: https://github.com/jax-ml/jax/releases/tag/jaxlib-v0.4.38

- python312Packages.jax: 0.4.28 -> 0.4.36
Diff: https://github.com/jax-ml/jax/compare/jax-v0.4.28...jax-v0.4.38
Changelog: https://github.com/jax-ml/jax/releases/tag/jax-v0.4.38


- Add binaries for aarch64-linux

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

natsukium committed:
python312Packages.jax: 0.4.28 -> 0.4.38

Diff: https://github.com/jax-ml/jax/compare/jax-v0.4.28...jax-v0.4.38

Changelog: https://github.com/jax-ml/jax/releases/tag/jax-v0.4.38

natsukium committed:
python312Packages.jaxlib: use jaxlib-bin by default

Thanks to bazel, it is difficult to source build with our current
resources.

natsukium commented on pkgs/development/python-modules/jaxlib/bin.nix:
```diff
@@ -68,96 +68,102 @@ let
-    "cuda12.2-3.11" = fetchurl {
-      url = "https://storage.googleapis.com/jax-releases/cuda12/jaxlib-${version}+cuda12.cudnn89-cp311-cp311-manylinux2014_x86_64.whl";
-      hash = "sha256-QO2WSOzmJ48VaCha596mELiOfPsAGLpGctmdzcCHE/o=";
+    "x86_64-linux" = fetchurl {
```
TODO:
Since CUDA support is now available as a plug-in system, it needs to be added to the dependencies as a separate package.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363130`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 24 packages marked as broken and skipped:</summary>
  <ul>
    <li>piper-train</li>
    <li>piper-train.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.elegy</li>
    <li>python312Packages.elegy.dist</li>
    <li>python312Packages.objax</li>
    <li>python312Packages.objax.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.tf2onnx</li>
    <li>python312Packages.tf2onnx.dist</li>
    <li>python312Packages.treeo</li>
    <li>python312Packages.treeo.dist</li>
    <li>python312Packages.treex</li>
    <li>python312Packages.treex.dist</li>
    <li>python313Packages.treeo</li>
    <li>python313Packages.treeo.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 171 packages failed to build:</summary>
  <ul>
    <li>beets</li>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
    <li>exo</li>
    <li>exo.dist</li>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>pianotrans</li>
    <li>pianotrans.dist</li>
    <li>pitivi</li>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.atomman</li>
    <li>python312Packages.atomman.dist</li>
    <li>python312Packages.awkward</li>
    <li>python312Packages.awkward-pandas</li>
    <li>python312Packages.awkward-pandas.dist</li>
    <li>python312Packages.awkward.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.boost-histogram</li>
    <li>python312Packages.boost-histogram.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.coffea</li>
    <li>python312Packages.coffea.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.correctionlib</li>
    <li>python312Packages.correctionlib.dist</li>
    <li>python312Packages.dask-awkward</li>
    <li>python312Packages.dask-awkward.dist</li>
    <li>python312Packages.dask-histogram</li>
    <li>python312Packages.dask-histogram.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.flowmc</li>
    <li>python312Packages.flowmc.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.hist</li>
    <li>python312Packages.hist.dist</li>
    <li>python312Packages.islpy</li>
    <li>python312Packages.islpy.dist</li>
    <li>python312Packages.jaxlibWithCuda</li>
    <li>python312Packages.jaxlibWithCuda.dist</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.librosa</li>
    <li>python312Packages.librosa.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.loopy</li>
    <li>python312Packages.loopy.dist</li>
    <li>python312Packages.manifold3d</li>
    <li>python312Packages.manifold3d.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.mplhep</li>
    <li>python312Packages.mplhep.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.nanobind</li>
    <li>python312Packages.nanobind.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.nutpie</li>
    <li>python312Packages.nutpie.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
    <li>python312Packages.oryx</li>
    <li>python312Packages.oryx.dist</li>
    <li>python312Packages.phonopy</li>
    <li>python312Packages.phonopy.dist</li>
    <li>python312Packages.piano-transcription-inference</li>
    <li>python312Packages.piano-transcription-inference.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.pyopencl</li>
    <li>python312Packages.pyopencl.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.reikna</li>
    <li>python312Packages.reikna.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python312Packages.soxr</li>
    <li>python312Packages.soxr.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
    <li>python312Packages.tensorflowWithCuda</li>
    <li>python312Packages.tensorflowWithCuda.dist</li>
    <li>python312Packages.tinygrad</li>
    <li>python312Packages.tinygrad.dist</li>
    <li>python312Packages.torch-audiomentations</li>
    <li>python312Packages.torch-audiomentations.dist</li>
    <li>python312Packages.torchcrepe</li>
    <li>python312Packages.torchcrepe.dist</li>
    <li>python312Packages.torchlibrosa</li>
    <li>python312Packages.torchlibrosa.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>python312Packages.uproot</li>
    <li>python312Packages.uproot.dist</li>
    <li>python312Packages.vector</li>
    <li>python312Packages.vector.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.dm-haiku</li>
    <li>python313Packages.dm-haiku.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.jaxlibWithCuda</li>
    <li>python313Packages.jaxlibWithCuda.dist</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxlib (python312Packages.jaxlib-bin ,python312Packages.jaxlibWithoutCuda)</li>
    <li>python312Packages.jaxlib.dist (python312Packages.jaxlib-bin.dist ,python312Packages.jaxlibWithoutCuda.dist)</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxlib (python313Packages.jaxlib-bin ,python313Packages.jaxlibWithoutCuda)</li>
    <li>python313Packages.jaxlib.dist (python313Packages.jaxlib-bin.dist ,python313Packages.jaxlibWithoutCuda.dist)</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 24 packages marked as broken and skipped:</summary>
  <ul>
    <li>piper-train</li>
    <li>piper-train.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.elegy</li>
    <li>python312Packages.elegy.dist</li>
    <li>python312Packages.objax</li>
    <li>python312Packages.objax.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.tf2onnx</li>
    <li>python312Packages.tf2onnx.dist</li>
    <li>python312Packages.treeo</li>
    <li>python312Packages.treeo.dist</li>
    <li>python312Packages.treex</li>
    <li>python312Packages.treex.dist</li>
    <li>python313Packages.treeo</li>
    <li>python313Packages.treeo.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 171 packages failed to build:</summary>
  <ul>
    <li>beets</li>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
    <li>exo</li>
    <li>exo.dist</li>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>pianotrans</li>
    <li>pianotrans.dist</li>
    <li>pitivi</li>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.atomman</li>
    <li>python312Packages.atomman.dist</li>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.awkward</li>
    <li>python312Packages.awkward-pandas</li>
    <li>python312Packages.awkward-pandas.dist</li>
    <li>python312Packages.awkward.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.boost-histogram</li>
    <li>python312Packages.boost-histogram.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.coffea</li>
    <li>python312Packages.coffea.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.correctionlib</li>
    <li>python312Packages.correctionlib.dist</li>
    <li>python312Packages.dask-awkward</li>
    <li>python312Packages.dask-awkward.dist</li>
    <li>python312Packages.dask-histogram</li>
    <li>python312Packages.dask-histogram.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.flowmc</li>
    <li>python312Packages.flowmc.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.hist</li>
    <li>python312Packages.hist.dist</li>
    <li>python312Packages.islpy</li>
    <li>python312Packages.islpy.dist</li>
    <li>python312Packages.jaxlibWithCuda</li>
    <li>python312Packages.jaxlibWithCuda.dist</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.librosa</li>
    <li>python312Packages.librosa.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.loopy</li>
    <li>python312Packages.loopy.dist</li>
    <li>python312Packages.manifold3d</li>
    <li>python312Packages.manifold3d.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.mplhep</li>
    <li>python312Packages.mplhep.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.nanobind</li>
    <li>python312Packages.nanobind.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.nutpie</li>
    <li>python312Packages.nutpie.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
    <li>python312Packages.oryx</li>
    <li>python312Packages.oryx.dist</li>
    <li>python312Packages.phonopy</li>
    <li>python312Packages.phonopy.dist</li>
    <li>python312Packages.piano-transcription-inference</li>
    <li>python312Packages.piano-transcription-inference.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.pyopencl</li>
    <li>python312Packages.pyopencl.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.reikna</li>
    <li>python312Packages.reikna.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python312Packages.soxr</li>
    <li>python312Packages.soxr.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
    <li>python312Packages.tinygrad</li>
    <li>python312Packages.tinygrad.dist</li>
    <li>python312Packages.torch-audiomentations</li>
    <li>python312Packages.torch-audiomentations.dist</li>
    <li>python312Packages.torchcrepe</li>
    <li>python312Packages.torchcrepe.dist</li>
    <li>python312Packages.torchlibrosa</li>
    <li>python312Packages.torchlibrosa.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>python312Packages.uproot</li>
    <li>python312Packages.uproot.dist</li>
    <li>python312Packages.vector</li>
    <li>python312Packages.vector.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.dm-haiku</li>
    <li>python313Packages.dm-haiku.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.jaxlibWithCuda</li>
    <li>python313Packages.jaxlibWithCuda.dist</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxlib (python312Packages.jaxlib-bin ,python312Packages.jaxlibWithoutCuda)</li>
    <li>python312Packages.jaxlib.dist (python312Packages.jaxlib-bin.dist ,python312Packages.jaxlibWithoutCuda.dist)</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxlib (python313Packages.jaxlib-bin ,python313Packages.jaxlibWithoutCuda)</li>
    <li>python313Packages.jaxlib.dist (python313Packages.jaxlib-bin.dist ,python313Packages.jaxlibWithoutCuda.dist)</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 74 packages marked as broken and skipped:</summary>
  <ul>
    <li>minari</li>
    <li>minari.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxlib</li>
    <li>python312Packages.jaxlib-bin</li>
    <li>python312Packages.jaxlib-bin.dist</li>
    <li>python312Packages.jaxlib.dist</li>
    <li>python312Packages.jaxlibWithoutCuda</li>
    <li>python312Packages.jaxlibWithoutCuda.dist</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.treeo</li>
    <li>python312Packages.treeo.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.dm-haiku</li>
    <li>python313Packages.dm-haiku.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxlib</li>
    <li>python313Packages.jaxlib-bin</li>
    <li>python313Packages.jaxlib-bin.dist</li>
    <li>python313Packages.jaxlib.dist</li>
    <li>python313Packages.jaxlibWithoutCuda</li>
    <li>python313Packages.jaxlibWithoutCuda.dist</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.minari</li>
    <li>python313Packages.minari.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
    <li>python313Packages.treeo</li>
    <li>python313Packages.treeo.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 24 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.elegy</li>
    <li>python312Packages.elegy.dist</li>
    <li>python312Packages.loopy</li>
    <li>python312Packages.loopy.dist</li>
    <li>python312Packages.objax</li>
    <li>python312Packages.objax.dist</li>
    <li>python312Packages.pyopencl</li>
    <li>python312Packages.pyopencl.dist</li>
    <li>python312Packages.reikna</li>
    <li>python312Packages.reikna.dist</li>
    <li>python312Packages.tf2onnx</li>
    <li>python312Packages.tf2onnx.dist</li>
    <li>python312Packages.treeo</li>
    <li>python312Packages.treeo.dist</li>
    <li>python312Packages.treex</li>
    <li>python312Packages.treex.dist</li>
    <li>python313Packages.treeo</li>
    <li>python313Packages.treeo.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 144 packages failed to build:</summary>
  <ul>
    <li>beets</li>
    <li>beets-unstable</li>
    <li>beets-unstable.dist</li>
    <li>beets-unstable.doc</li>
    <li>beets-unstable.man</li>
    <li>beets.dist</li>
    <li>beets.doc</li>
    <li>beets.man</li>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>pianotrans</li>
    <li>pianotrans.dist</li>
    <li>python312Packages.array-api-compat</li>
    <li>python312Packages.array-api-compat.dist</li>
    <li>python312Packages.atomman</li>
    <li>python312Packages.atomman.dist</li>
    <li>python312Packages.augmax</li>
    <li>python312Packages.augmax.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.blackjax</li>
    <li>python312Packages.blackjax.dist</li>
    <li>python312Packages.boost-histogram</li>
    <li>python312Packages.boost-histogram.dist</li>
    <li>python312Packages.chex</li>
    <li>python312Packages.chex.dist</li>
    <li>python312Packages.coffea</li>
    <li>python312Packages.coffea.dist</li>
    <li>python312Packages.dask-awkward</li>
    <li>python312Packages.dask-awkward.dist</li>
    <li>python312Packages.dask-histogram</li>
    <li>python312Packages.dask-histogram.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.dm-haiku</li>
    <li>python312Packages.dm-haiku.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.equinox</li>
    <li>python312Packages.equinox.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.flowmc</li>
    <li>python312Packages.flowmc.dist</li>
    <li>python312Packages.hist</li>
    <li>python312Packages.hist.dist</li>
    <li>python312Packages.islpy</li>
    <li>python312Packages.islpy.dist</li>
    <li>python312Packages.jax</li>
    <li>python312Packages.jax.dist</li>
    <li>python312Packages.jaxopt</li>
    <li>python312Packages.jaxopt.dist</li>
    <li>python312Packages.jmp</li>
    <li>python312Packages.jmp.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.librosa</li>
    <li>python312Packages.librosa.dist</li>
    <li>python312Packages.lineax</li>
    <li>python312Packages.lineax.dist</li>
    <li>python312Packages.manifold3d</li>
    <li>python312Packages.manifold3d.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.mplhep</li>
    <li>python312Packages.mplhep.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.nanobind</li>
    <li>python312Packages.nanobind.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.optax</li>
    <li>python312Packages.optax.dist</li>
    <li>python312Packages.optax.testsout</li>
    <li>python312Packages.optimistix</li>
    <li>python312Packages.optimistix.dist</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
    <li>python312Packages.oryx</li>
    <li>python312Packages.oryx.dist</li>
    <li>python312Packages.phonopy</li>
    <li>python312Packages.phonopy.dist</li>
    <li>python312Packages.piano-transcription-inference</li>
    <li>python312Packages.piano-transcription-inference.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python312Packages.soxr</li>
    <li>python312Packages.soxr.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
    <li>python312Packages.torch-audiomentations</li>
    <li>python312Packages.torch-audiomentations.dist</li>
    <li>python312Packages.torchcrepe</li>
    <li>python312Packages.torchcrepe.dist</li>
    <li>python312Packages.torchlibrosa</li>
    <li>python312Packages.torchlibrosa.dist</li>
    <li>python312Packages.treescope</li>
    <li>python312Packages.treescope.dist</li>
    <li>python312Packages.vector</li>
    <li>python312Packages.vector.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.augmax.dist</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.chex</li>
    <li>python313Packages.chex.dist</li>
    <li>python313Packages.dm-haiku</li>
    <li>python313Packages.dm-haiku.dist</li>
    <li>python313Packages.equinox</li>
    <li>python313Packages.equinox.dist</li>
    <li>python313Packages.jax</li>
    <li>python313Packages.jax.dist</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.jmp</li>
    <li>python313Packages.jmp.dist</li>
    <li>python313Packages.lineax</li>
    <li>python313Packages.lineax.dist</li>
    <li>python313Packages.minari</li>
    <li>python313Packages.minari.dist</li>
    <li>python313Packages.optax</li>
    <li>python313Packages.optax.dist</li>
    <li>python313Packages.optax.testsout</li>
    <li>python313Packages.optimistix</li>
    <li>python313Packages.optimistix.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.jaxlib (python312Packages.jaxlib-bin ,python312Packages.jaxlibWithoutCuda)</li>
    <li>python312Packages.jaxlib.dist (python312Packages.jaxlib-bin.dist ,python312Packages.jaxlibWithoutCuda.dist)</li>
    <li>python313Packages.jaxlib (python313Packages.jaxlib-bin ,python313Packages.jaxlibWithoutCuda)</li>
    <li>python313Packages.jaxlib.dist (python313Packages.jaxlib-bin.dist ,python313Packages.jaxlibWithoutCuda.dist)</li>
  </ul>
</details>

GaetanLepage commented on pkgs/development/python-modules/jaxlib/bin.nix:
```diff
@@ -168,10 +174,10 @@ buildPythonPackage {
 
   disabled =
```
```suggestion
  disabled = pythonOlder "3.10";
```
I am not sure that keeping an upper bound makes a ton of sense.
Actually, I would be in favor of removing the `disabled` attribute entirely.

natsukium replied:
I'm not aggressively trying to remove it, but it seems a bit redundant here.

GaetanLepage commented on pkgs/development/python-modules/jaxlib/bin.nix:
```diff
@@ -231,15 +238,18 @@ buildPythonPackage {
     license = licenses.asl20;
     maintainers = with maintainers; [ samuela ];
     platforms = [
+      "aarch64-linux"
       "aarch64-darwin"
       "x86_64-linux"
       "x86_64-darwin"
     ];
```
```suggestion
```
Shouldn't we now remove the `platforms` attribute entirely as they are now all supported (which is the default) ?

GaetanLepage commented on pkgs/development/python-modules/jax/default.nix:
```diff
@@ -27,20 +26,20 @@ let
 
-  disabled = pythonOlder "3.9";
+  disabled = pythonOlder "3.10";
 
```
Same, I would get rid of that.

GaetanLepage reviewed (changes requested):
Thanks a lot for tackling this :)

stephen-huan commented on pkgs/development/python-modules/jax/default.nix:
```diff
@@ -49,16 +48,16 @@ buildPythonPackage rec {
   # jaxlib is _not_ included in propagatedBuildInputs because there are
   # different versions of jaxlib depending on the desired target hardware. The
   # JAX project ships separate wheels for CPU, GPU, and TPU.
-  propagatedBuildInputs = [
+  dependencies = [
+    jaxlib
```
Is this comment and https://github.com/NixOS/nixpkgs/pull/156808 no longer relevant now that jax depends on jaxlib directly with the [CUDA plugin change](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-30-june-18-2024) in `0.4.30`? I was sitting on a PR removing jaxlib from {dm-haiku, equinox, flax, optax, etc.} but I guess it's no longer relevant. And should packages depending on `jax` also put `jaxlib` in their `dependencies` after this PR or is that no longer necessary as `jax` itself now depends on `jaxlib`?

Thanks for tackling this by the way!

samuela replied:
> Is this comment and https://github.com/NixOS/nixpkgs/pull/156808 no longer relevant now that jax depends on jaxlib directly with the [CUDA plugin change](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-30-june-18-2024) in 0.4.30?

That is also my understanding

> I was sitting on a PR removing jaxlib from {dm-haiku, equinox, flax, optax, etc.} but I guess it's no longer relevant. And should packages depending on jax also put jaxlib in their dependencies after this PR or is that no longer necessary as jax itself now depends on jaxlib?

I agree that this situation is a bit confusing currently. I propose that we merge this PR along with written guidance to maintainers on migrating.

GaetanLepage replied:
Yes, from my understanding, we will be able to get rid of specifying `jaxlib` in downstream packages.

natsukium replied:
I'll delete this comment as it causes confusion.

> I propose that we merge this PR along with written guidance to maintainers on migrating.

Do you mean to add in the release notes that it is not necessary to explicitly add `jaxlib` as a dependency for packages that depend on `jax`?

natsukium replied:
I added the information to the release note.
https://github.com/NixOS/nixpkgs/pull/363130/commits/b69c1a410ef9dde6999e3321a55133d776bec347

GaetanLepage commented:
Trying to "build" `jaxlib` with `cudaSupport` enabled gives:
```
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: jaxlib jaxlib.cpu_feature_guard jaxlib.xla_client
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'jaxlib.cpu_feature_guard'
```

samuela commented:
> Trying to "build" jaxlib with cudaSupport enabled gives:

Looks like we can just remove it from `pythonImportsChecks` then. I forget why we had it there in the first place... perhaps that's worth figuring out

GaetanLepage commented:
@natsukium ping.
Do you need any further help with this ? It would be great to have it merged in the current cycle.

GaetanLepage commented:
0.4.38 is out: https://github.com/jax-ml/jax/releases/tag/jax-v0.4.38

natsukium commented:
I had been away from home for a while on business, and I was sick in bed.
Let's clean up the task.

mweinelt commented:
We're about to merge staging-next today or tomorrow I believe.

natsukium commented:
Sorry for the slow communication. I'm back.
@GaetanLepage, thanks for trying to move forward.
I have a patch for the cuda-plugin improvement on my machine, but unfortunately it still can't support CUDA.
However, this PR has already become a blocker for all packages that depend on jax, so I think it is better to merge it once.

natsukium committed:
python312Packages.jaxlib-bin: 0.4.28 -> 0.4.38

Diff: jax-ml/jax@jaxlib-v0.4.28...jaxlib-v0.4.38

Changelog: https://github.com/jax-ml/jax/releases/tag/jaxlib-v0.4.38

natsukium committed:
python312Packages.jaxlib: update prefetch.sh

natsukium committed:
python312Packages.jax-cuda12-pjrt: init at 0.4.38

natsukium committed:
python312Packages.jax-cuda12-plugin: init at 0.4.38

natsukium committed:
python312Packages.nifty8: don't override jaxlib

natsukium commented:
I think it is ready for the merge.

GaetanLepage reviewed (approved):
Let's go
